search_queries_interval.go

```diff
@@ -0,0 +1,43 @@
+// Copyright 2012-present Oliver Eilhard. All rights reserved.
+// Use of this source code is governed by a MIT-license.
+// See http://olivere.mit-license.org/license.txt for details.
+
+package elastic
+
+// IntervalRule represents the generic matching rule interface. Matching Rule is
+// just a Query, but may be used only inside IntervalQuery. An extra method is added
+// just to separate *Rule objects from other queries
+type IntervalRule interface {
+	Query
+
+	// IsIntervalRule is never actually called, and is used just for Rule to differ from standard Query
+	IsIntervalRule() bool
```

Comment:
If it's only used internally, you could mark it as unexported: `isIntervalRule() bool`?